api/agent/agent.go

```diff
@@ -807,6 +742,107 @@ func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state
 	logger.WithError(err).Info("hot function terminated")
 }
 
+// runHotReq enqueues a free slot to slot queue manager and watches various timers and the consumer until
+// the slot is consumed. A return value of false means, the container should shutdown and no subsequent
+// calls should be made to this function.
+func (a *agent) runHotReq(ctx context.Context, call *call, state ContainerState, logger logrus.FieldLogger, cookie drivers.Cookie, slot *hotSlot) bool {
+
+	var err error
+	isFrozen := false
+
+	freezeTimer := time.NewTimer(a.freezeIdleMsecs)
+	idleTimer := time.NewTimer(time.Duration(call.IdleTimeout) * time.Second)
+	ejectTicker := time.NewTicker(a.ejectIdleMsecs)
```

Comment:
i gave this a solid 30 seconds of thought, but I can't come up with why this one is a ticker and the other 2 are timers. tickers may drop ticks if the receiver is slow, which isn't something we want exactly I don't think (say, freeze latency puts us past the eject timeout, do we want to skip ejection due to this?). a ticker also will send > 1 time, whereas this function pertains to one instance of acquiring a slot or $alternative (alternative being: shutdown, eject, idle). anyway... ?

Reply:
I think missing ticks due to freeze latency is OK. It's not really an idleness due to no processing. It is a ticker because we want to check if we can eject it every eg 1 seconds if it was idle.

Reply:
ah, that makes sense.